### ISSUE TITLE
The model is converted to onnx format using dynamic batch precision collapse

### ISSUE THREAD

**Issue by ffxxjj** [onnx/onnx#6192]
Here is the code I used to convert the model to onnx format, where I enabled dynamic batch

def export_onnx(onnx_model_name,net,input_dicts:dict,output_name:list,dynamic=True,simplify=False):
    '''
        input_dicts = {"input1":(1,3,224,224),"input2":(1,3,224,224),...}
        output_name = {"output1","output2",...}

    '''
    inp = (torch.randn(i) for i in input_dicts.values())
    # inp.to('cuda')
    # net.to('cuda')
    # net=torch.jit.script(net)
    net.eval()
    # generate ONNX model
    input_name = [i for i in input_dicts]
    all_names = input_name+output_name
    dynamic_axes={}
    for i in all_names:
        dynamic_axes[i]={0:'batch'}
    
    # Export the model
    torch.onnx.export(net,               # model being run
                    tuple(inp),                         # model input (or a tuple for multiple inputs)
                    onnx_model_name,   # where to save the model (can be a file or file-like object)
                    export_params=True,        # store the trained parameter weights inside the model file
                    opset_version=17,          # the ONNX version to export the model to
                    #   do_constant_folding=True,  # whether to execute constant folding for optimization
                    input_names = [i for i in input_name],   # the model's input names
                    output_names = [i for i in output_name], # the model's output names
                    dynamic_axes=dynamic_axes if dynamic else None,
                    )
    print(dynamic_axes)
    onnx_model = onnx.load(onnx_model_name)
    onnx.checker.check_model(onnx_model)

    # 验证精度差异
    ort_session = onnxruntime.InferenceSession(onnx_model_name)
    def to_numpy(tensor):
        return tensor.detach().cpu().numpy() if tensor.requires_grad else tensor.cpu().numpy()
    # compute ONNX Runtime output prediction
    ort_inputs = {i.name:to_numpy(torch.randn(input_dicts[i.name])) for i in ort_session.get_inputs()}
    ort_outs = ort_session.run(None, ort_inputs)
    net_input=[torch.tensor(i) for i in ort_inputs.values()]
    torch_out = net(*net_input)
    # compare ONNX Runtime and PyTorch results
    np.testing.assert_allclose(to_numpy(torch_out[0]), ort_outs[0][0], rtol=1e-04, atol=1e-05)
    # np.testing.assert_allclose(to_numpy(torch_out[1]), ort_outs[1], rtol=1e-03, atol=1e-05)

    print("Exported model has been tested with ONNXRuntime, and the result looks good!")

    # Simplify 简化onnx模型
    if simplify:
        try:
            model_onnx, check = onnxsim.simplify(onnx_model_name)
            assert check, 'assert check failed'
            onnx.save(model_onnx, onnx_model_name)
        except Exception as e:
            print(f'simplifier failure: {e}')

input_dicts = {
    "input":(1,3,448,448),
}

export_onnx(onnx_model_name = 'model.onnx',net = net,input_dicts=input_dicts,output_name=['output'],dynamic=True,simplify=True)

But when I use the converted model to reason, the accuracy is only 15%, what causes this problems?

**Comment by xadupre** [comment]
It is difficult to answer without knowing the model. Could you try with simplify=False just to make sure this function is not involved. Then what do you mean by 15%? It is 15% of a batch of observations are not equal to the expected outputs? So it means 85% are correct and the model is probably correctly converted.

**Comment by xadupre** [comment]
So the model is probably right since 85% of the observations are correctly classified by onnx. Is it possible to check the highest probabilities instead of the label? Sometimes, it two classes have similar probabilities, a small difference may change the top one.

**Comment by ffxxjj** [comment]
I mean 85% of the observations are misclassified, but I use batchsize=1 to deduce something close to the original model, and I can't understand what's wrong with that

**Comment by xadupre** [comment]
Can you check the discrepancies for the probabilities on the failing observations?

**Comment by ffxxjj** [comment]
Do you mean the difference between the probability of failure predicted by the onnx model and the predicted value of the original model?

**Comment by xadupre** [comment]
between the probability of predicted by the onnx model and the predicted value of the original model for the observations which fail the first test.

**Comment by ffxxjj** [comment]
Okay, thanks. I'll check it out
